controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -0,0 +1,226 @@
+import static io.pravega.shared.MetricsNames.CREATE_STREAM_FAILED;
+import static io.pravega.shared.MetricsNames.CREATE_STREAM_LATENCY;
+import static io.pravega.shared.MetricsNames.DELETE_STREAM;
+import static io.pravega.shared.MetricsNames.DELETE_STREAM_FAILED;
+import static io.pravega.shared.MetricsNames.DELETE_STREAM_LATENCY;
+import static io.pravega.shared.MetricsNames.OPEN_TRANSACTIONS;
+import static io.pravega.shared.MetricsNames.RETENTION_FREQUENCY;
+import static io.pravega.shared.MetricsNames.SEAL_STREAM;
+import static io.pravega.shared.MetricsNames.SEAL_STREAM_FAILED;
+import static io.pravega.shared.MetricsNames.SEAL_STREAM_LATENCY;
+import static io.pravega.shared.MetricsNames.SEGMENTS_COUNT;
+import static io.pravega.shared.MetricsNames.SEGMENTS_MERGES;
+import static io.pravega.shared.MetricsNames.SEGMENTS_SPLITS;
+import static io.pravega.shared.MetricsNames.TRUNCATE_STREAM;
+import static io.pravega.shared.MetricsNames.TRUNCATE_STREAM_FAILED;
+import static io.pravega.shared.MetricsNames.TRUNCATE_STREAM_LATENCY;
+import static io.pravega.shared.MetricsNames.UPDATE_STREAM;
+import static io.pravega.shared.MetricsNames.UPDATE_STREAM_FAILED;
+import static io.pravega.shared.MetricsNames.UPDATE_STREAM_LATENCY;
+import static io.pravega.shared.MetricsNames.nameFromStream;
+
+/**
+ * Class to encapsulate the logic to report Controller service metrics for Streams.
+ */
+public final class StreamMetrics extends AbstractControllerMetrics implements AutoCloseable {
+
+    private final OpStatsLogger createStreamLatency = STATS_LOGGER.createStats(CREATE_STREAM_LATENCY);
+    private final OpStatsLogger deleteStreamLatency = STATS_LOGGER.createStats(DELETE_STREAM_LATENCY);
+    private final OpStatsLogger sealStreamLatency = STATS_LOGGER.createStats(SEAL_STREAM_LATENCY);
+    private final OpStatsLogger updateStreamLatency = STATS_LOGGER.createStats(UPDATE_STREAM_LATENCY);
+    private final OpStatsLogger truncateStreamLatency = STATS_LOGGER.createStats(TRUNCATE_STREAM_LATENCY);
+
+    /**
+     * This method increments the global and Stream-specific counters of Stream creations, initializes other
+     * stream-specific metrics and reports the latency of the operation.
+     *
+     * @param scope             Scope.
+     * @param streamName        Name of the Stream.
+     * @param minNumSegments    Initial number of segments for the Stream.
+     * @param latency           Latency of the createStream operation.
+     */
+    public void createStream(String scope, String streamName, int minNumSegments, Duration latency) {
+        DYNAMIC_LOGGER.incCounterValue(CREATE_STREAM, 1);
+        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(OPEN_TRANSACTIONS, scope, streamName), 0);
+        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName), minNumSegments);
+        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, streamName), 0);
```

Comment:
instead of creating new counter why are you calling increment counter value with 0? is it because we do not have createCounter method in Dynamic logger?

Reply:
Yes, we only have `incCounter()` and `updateCounter()` methods. In fact, this is legacy code moved to this new class.